calc/src/mechanics/util.ts

```diff
@@ -109,11 +109,13 @@ export function getFinalSpeed(gen: Generation, pokemon: Pokemon, field: Field, s
   (pokemon.hasAbility('Chlorophyll') && weather.includes('Sun')) ||
   (pokemon.hasAbility('Swift Swim') && weather.includes('Rain')) ||
   (pokemon.hasAbility('Slush Rush') && ['Hail', 'Snow'].includes(weather)) ||
-  (pokemon.hasAbility('Surge Surfer') && (terrain === 'Electric') || pokemon.hasAbility('Levitate', 'DragonFly')) ||
```

Comment:
thats a recent oopsie whoospies, i mean if the pokemon has Surge surfer and maybe is flying or the terrain is electric